pkg/services/folder/folderimpl/folder.go

```diff
@@ -555,7 +557,7 @@ func (s *Service) Create(ctx context.Context, cmd *folder.CreateFolderCommand) (
 	dashFolder := dashboards.NewDashboardFolder(cmd.Title)
 	dashFolder.OrgID = cmd.OrgID
 
-	if s.features.IsEnabled(ctx, featuremgmt.FlagNestedFolders) && cmd.ParentUID != "" {
```

Comment:
When the feature flag is disabled and provisioning dashboards in nested folders, a folder is stored without parent folder_uid in `dashboards` table.

So when provisioning service checks by title & folder_uid if such a folder exists (in `getOrCreateFolderByTitle` method), it fails to find a match and attempts to create it again even thought it already exists.

Reply:
I'm leaning towards it's better to not create nested folders if the flag is not set therefore I would suggest to better hide my changes behind the feature toggle.

Reply:
draft for this is in branch https://github.com/grafana/grafana/tree/provisioning-with-featToggle